otter/convergence/service.py

```diff
@@ -0,0 +1,58 @@
+"""Converger service"""
+
+from effect import Effect
+from effect.twisted import perform
+
+from twisted.application.service import Service
+
+from otter.constants import CONVERGENCE_LOCK_PATH
+from otter.convergence.composition import (
+    execute_convergence, get_desired_group_state)
+from otter.http import TenantScope
+from otter.util.deferredutils import with_lock
+
+
+class Converger(Service, object):
+    """Converger service"""
+
+    def __init__(self, reactor, kz_client, dispatcher):
+        self._reactor = reactor
+        self._kz_client = kz_client
+        self._dispatcher = dispatcher
+
+    def _get_lock(self, group_id):
+        """Get a ZooKeeper-backed lock for converging the group."""
+        path = CONVERGENCE_LOCK_PATH.format(group_id=group_id)
+        return self._kz_client.Lock(path)
+
+    def converge(self, log, tenant_id, group_id, desired, launch_config,
```

Comment:
Do you think this can be called `set_desired_state` or something similar? We already have lot of `converge` functions in our code :)